### PR TITLE
feat: financial DLP patterns with checksum validation

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary

- Add credit card (Luhn) and IBAN (mod-97) detection to the DLP scanner with post-match checksum validation that eliminates 90-99% of false positives
- Introduce `Validator` field on `DLPPattern` and `validate` function on `compiledPattern` — extensible architecture for future validated pattern types
- Credit card regex covers Visa, Mastercard (including 2221-2720 series), Amex, Discover, and JCB. IBAN covers all 80+ countries
- ABA routing number validator available as opt-in config (not in defaults due to higher FP rate)
- DLP pattern count 44 → 46. All 7 config presets updated. 70 new tests.

## Test plan

- [x] All in-repo Go tests pass (6,675 total, 70 new)
- [x] Validator unit tests: Luhn, mod-97, ABA with valid/invalid/edge cases
- [x] URL DLP integration: valid cards blocked, invalid Luhn allowed (FP rejection)
- [x] Text DLP integration: valid IBANs detected, invalid mod-97 rejected
- [x] ABA opt-in: config wiring verified, not in defaults
- [x] Config validation: invalid validator name rejected, valid names accepted
- [x] All 7 presets parse and load correctly
- [x] Coverage: validate.go 94.7-100%, scanner package 96.7%

## Summary by CodeRabbit

* **New Features**
  - DLP coverage increased from 44 to 46 built-in patterns
  - Added Credit Card and IBAN detections with checksum validation (Luhn, MOD‑97) and optional ABA validator
  - Per-pattern checksum validators can be enabled to reduce false positives

* **Documentation**
  - Updated README, configuration docs, guides, benchmarks, and mappings to reflect 46 patterns and validator usage (including sample health response)

* **Tests**
  - Expanded DLP/scanner tests covering new patterns, validators, regression cases

### CODE REVIEW

File: internal/scanner/validate.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		if isASCIIDigit(c) {
+			remainder = (remainder*10 + int(c-'0')) % 97
+		} else if isASCIILetter(c) {
+			remainder = (remainder*100 + int(c-'A') + 10) % 97
+		} else {
+			return false // unexpected character in BBAN
+		}
+	}
+
+	return remainder == 1
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/validate.go:156]: Fixed in commit 1522e38. `validateMod97` now uses `ibanCountryLengths` map (86 countries with exact expected lengths). A 15-char DE IBAN is rejected because German IBANs must be 22 chars. Fake country codes (ZZ, XX) are also rejected. Tests added for both cases.

File: internal/config/config.go [hunk 2]
```diff
@@ -2564,6 +2579,14 @@ func Defaults() *Config {
...
 				// Case-insensitive matching is added automatically by scanner.New() via (?i) prefix.
 				{Name: "Credential in URL", Regex: `\b(?:password|passwd|secret|token|apikey|api_key|api-key)\s*=\s*[^\s&]{4,}`, Severity: "high"},
+
+				// Financial identifiers — validated with post-match checksums to minimize
+				// false positives. Luhn drops ~90% of random 16-digit matches, mod-97
+				// drops ~99% of random IBAN-format matches. ABA is not in defaults due
+				// to high FP rate (~10% of random 9-digit numbers pass checksum); users
+				// can add it via config with validator: "aba".
+				{Name: "Credit Card Number", Regex: `\b(?:4\d{3}|5[1-5]\d{2}|2(?:2(?:2[1-9]|[3-9]\d)|[3-6]\d{2}|7[01]\d|720)|3[47]\d{2}|6(?:011|[45]\d{2})|35(?:2[89]|[3-8]\d))(?:[- ]?\d){11,15}\b`, Severity: "medium", Validator: ValidatorLuhn},
+				{Name: "IBAN", Regex: `\b[A-Z]{2}\d{2}[A-Z0-9]{11,30}\b`, Severity: "medium", Validator: ValidatorMod97},
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/config/config.go:2602]: Intentional design choices for an exfiltration scanner (not a payment processor):

- **13-digit Visa:** Not issued since ~2000. Adding regex complexity for a dead format is not worth it.
- **Discover 622126-622925:** Intentionally excluded. Overlaps with UnionPay 62xx range, adding it creates more false positives than it catches.
- **6400-6439:** Discover network does use 644-649. The bot is incorrect that these are invalid.
- **Per-issuer length enforcement:** Luhn checksum already validates the actual number. A 16-digit string starting with 37 that passes Luhn is still worth flagging as a potential financial identifier in exfiltration context. The security gain from per-issuer length enforcement is zero (Luhn catches it), while the regex complexity cost is significant.

This is a DLP pattern in a network proxy, not a payment gateway. False negatives from missing dead formats are acceptable. False positives from slightly broad BIN ranges are caught by Luhn. The tradeoff favors simplicity.

File: internal/scanner/validate.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	case 4: // Visa: starts with 4, 16/19 digits (13-digit Visa retired in the '90s)
+		return n == 16 || n == 19
+	case 5: // Mastercard: 51-55, 16 digits
+		return digits[1] >= 1 && digits[1] <= 5 && n == 16
+	case 3: // Amex (34/37): 15 digits. JCB (3528-3589): 16 digits.
+		if digits[1] == 4 || digits[1] == 7 {
+			return n == 15 // Amex
+		}
+		if prefix4 >= 3528 && prefix4 <= 3589 {
+			return n == 16 // JCB
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/validate.go]: Fixed — JCB now accepts 16-19 digits per network spec.
